src/saleor_app/install.py

```diff
@@ -1,59 +1,59 @@
+from typing import Dict, Optional
 
-from saleor_app.conf import get_settings
 from saleor_app.errors import InstallAppError
-from saleor_app.graphql import GraphQLError, get_executor, get_saleor_api_url
-from saleor_app.mutations import CREATE_WEBHOOK
-from saleor_app.schemas.core import AppToken, DomainName, Url, WebhookData
+from saleor_app.saleor.exceptions import GraphQLError
+from saleor_app.saleor.mutations import CREATE_WEBHOOK
+from saleor_app.saleor.utils import get_client_for_app
+from saleor_app.schemas.core import AppToken, DomainName, WebhookData
+from saleor_app.schemas.manifest import Manifest
 
 logger = logging.getLogger(__name__)
 
 
 async def install_app(
-    domain: DomainName,
-    token: AppToken,
-    events: List[str],
-    target_url: Url,
-    save_app_data: Callable[[DomainName, WebhookData], Awaitable],
+    saleor_domain: DomainName,
+    auth_token: AppToken,
+    manifest: Manifest,
+    events: Optional[Dict[str, str]],
+    use_insecure_saleor_http: bool,
 ):
     alphabet = string.ascii_letters + string.digits
     secret_key = "".join(secrets.choice(alphabet) for _ in range(20))
 
-    api_url = get_saleor_api_url(domain)
-    executor = get_executor(host=api_url, auth_token=token)
-
-    settings = get_settings()
-
-    response, errors = await executor(
-        CREATE_WEBHOOK,
-        variables={
-            "input": {
-                "targetUrl": target_url,
-                "events": [event.upper() for event in events],
-                "name": settings.app_name,
-                "secretKey": secret_key,
-            }
-        },
-    )
+    schema = "http" if use_insecure_saleor_http else "https"
+
+    errors = []
+
+    async with get_client_for_app(
+        f"{schema}://{saleor_domain}", manifest=manifest, auth_token=auth_token
+    ) as saleor_client:
+        for target_url, event_types in events.items():
+            try:
+                response = await saleor_client.execute(
+                    CREATE_WEBHOOK,
+                    variables={
+                        "input": {
+                            "targetUrl": str(target_url),
+                            "events": [event.upper() for event in event_types],
+                            "name": f"{manifest.name}",
+                            "secretKey": secret_key,
+                        }
+                    },
+                )
+            except GraphQLError as exc:
+                errors.append(exc)
 
     if errors:
-        logger.warning("Webhook create mutation raised an error")
-        raise GraphQLError("Webhook create mutation raised an error")
-
-    webhook_error = response["data"]["webhookCreate"].get("errors")
-    if webhook_error:
-        logger.warning(
-            "Unable to finish installation of app for %s. Received error: %s",
-            domain,
-            webhook_error,
+        logger.error("Unable to finish installation of app for %s.", saleor_domain)
```

Comment:
We should figure out how to treat an installation when some of the requests will pass successfully and the rest fail. But I don't think this is a scope of this PR, I just raise it to keep in mind